### ISSUE TITLE
`find_repo` function can enter an infinite loop

### ISSUE THREAD

**Issue by smithx** [libgit2/libgit2#6524]
### Reproduction steps
call
```
git_buf out{};
git_repository_discover(&out, "\\\\?\\Volume{20693787-e361-42da-97c9-f66d2468f001}\\EFI\\Boot", 0, NULL);
```

### Expected behavior
function return

### Actual behavior
function doesn't return
subcall `find_repo` never [break from loop ](https://github.com/libgit2/libgit2/blob/main/src/libgit2/repository.c#L680-L747)

### Version of libgit2 (release number or SHA1)
'main' branch 8164b481

### Operating system(s) tested
Windows 10 Pro 22H2

**Comment by ethomson** [comment]
I've struggled here to reproduce this — I don't seem to have this problem on my Windows machine (using my volume GUID). I'll keep trying, but if there's any additional data to reproduce this, that would be helpful.

**Comment by smithx** [comment]
You don't need Windows machine to reproduce it. Just check the unit test from PR https://github.com/libgit2/libgit2/pull/6525

test function name is 'test_repo_discover__discovery_starting_at_unmounted_win_root_causes_no_hang'

**Comment by ethomson** [comment]
That test runs successfully on all my machines (macOS, Windows 11) _without_ any changes to the find repo function. Does it still fail for you without any changes?

**Comment by smithx** [comment]
You are absolutely right, the test will not fail with some 'uuid-volume' path. We need such a 'uuid-volume' path that will successfully pass [here: `git_fs_path_prettify(&path, start_path, NULL)`](https://github.com/libgit2/libgit2/blob/main/src/libgit2/repository.c#L753). 
If we go into the [loop ](https://github.com/libgit2/libgit2/blob/main/src/libgit2/repository.c#L776-L824) with `path.ptr` points to something like this `\\\\?\\Volume{11111111-2222-3333-4444-555555555555}\\EFI\\Boot`, we will never get out again
